ui/src/lib/nodes.tsx

```diff
@@ -98,10 +98,11 @@ export function useNodesStateSynced(nodeList) {
 
       // TOFIX: a node may be shadowed behind its parent, due to the order to render reactflow node, to fix this, comment out the following sorted method, which brings in a large overhead.
       setNodes(
-        Array.from(nodesMap.values()).sort(
-          (a: Node & { level }, b: Node & { level }) => a.level - b.level
-        )
+        Array.from(nodesMap.values())
+          .sort((a: Node & { level }, b: Node & { level }) => a.level - b.level)
+          .map((node) => ({ ...node, selected: getPod(node.id)?.selected }))
```

Comment:
The only ground truth source is `nodesMap`, which is shared by all users. We have to maintain an individual selected state for each user. Actually, we have already implemented de-selection.

Reply:
Oh, I got what you meant. I realized its cons just now: new selection change will not be applied if no update in the remote `nodesMap`, un-selecting a node by clicking on the pane is a very typical case in which no update is written into `nodesMap`, so it will not sync the selection change in time.

I have an idea about the possible solution: stop writing `selected` fields of nodes into `nodesMap`, keep a `selectedNode`  set in store, observe any update both on `nodesMap` and `selectedNode` to compose the complete data of nodes. But notice `nodes` in reactflow is just a list not map, which may cause some performance issue.